integration/sawtooth_integration/tests/test_identity.py

```diff
@@ -0,0 +1,399 @@
+        # From local configuration Carol is denied from XO, but is allowed all
+        # other transactor permissions, Mallory and Chuck are denied from all
+        # transactor permissions.
+
+        self.walter.set_pubkey_for_role(
+            "allow_carol_and_no_others",
+            "transactor.transaction_signer",
+            permit_keys=[self.carol.public_key, self.walter.public_key],
+            deny_keys=[self.alice.public_key, self.bob.public_key,
+                       self.dave.public_key])
+
+        self.assert_able_to_send(self.carol, Families.INTKEY)
+        self.assert_not_able_to_send(self.carol, Families.XO)
+
+        for t, f in [(self.alice, Families.INTKEY),
+                     (self.alice, Families.XO),
+                     (self.bob, Families.INTKEY),
+                     (self.bob, Families.XO),
+                     (self.dave, Families.INTKEY),
+                     (self.dave, Families.XO),
+                     (self.chuck, Families.INTKEY),
+                     (self.chuck, Families.XO),
+                     (self.mallory, Families.INTKEY),
+                     (self.mallory, Families.XO)]:
+            self.assert_not_able_to_send(t, f)
+
+        self.walter.set_pubkey_for_role(
+            "allow_all_transaction_signers",
+            "transactor.transaction_signer",
+            permit_keys=["*"],
+            deny_keys=[])
+
+        #
+        # transactor.transaction_signer.< tp_name > subsection
+        #
+        # From local configuration Dave is denied from being a batch_signer,
+        # Mallory and Chuck are denied being transactors.
+
+        self.walter.set_pubkey_for_role(
+            "deny_alice_from_xo_allow_bob",
+            "transactor.transaction_signer.xo",
+            permit_keys=[self.bob.public_key, self.dave.public_key],
+            deny_keys=[self.alice.public_key])
+
+        self.assert_able_to_send(self.bob, Families.XO)
+
+        for t, f in [(self.alice, Families.XO), (self.chuck, Families.XO),
+                     (self.mallory, Families.XO), (self.dave, Families.XO)]:
+            self.assert_not_able_to_send(t, f)
+
+        self.walter.set_pubkey_for_role(
+            "deny_bob_from_intkey_allow_dave_alice",
+            "transactor.transaction_signer.intkey",
+            permit_keys=[self.alice.public_key, self.dave.public_key],
+            deny_keys=[self.bob.public_key])
+
+        self.assert_able_to_send(self.alice, Families.INTKEY)
+
+        for t, f in [(self.bob, Families.INTKEY), (self.chuck, Families.INTKEY),
+                     (self.mallory, Families.INTKEY), (self.dave, Families.INTKEY)]:
+            self.assert_not_able_to_send(t, f)
+
+    def assert_able_to_send(self, transactor, family):
+        transactor.send(family)
+
+    def assert_able_to_send_batch(self, transactor, family, transactions):
+        transactor.send(family_name=family, transactions=transactions)
+
+    def assert_not_able_to_send(self, transactor, family):
```

Comment:
`Exception` should be more specific if possible.

Reply:
RestClient in integration_tools.py transforms any non 200, 201, or 202 http status into an Exception. https://github.com/hyperledger/sawtooth-core/blob/master/integration/sawtooth_integration/tests/integration_tools.py#L107. Failures from the permission_verifier for transactors get a 400.

Reply:
Hmm, that's not great. Is it possible to get the specific code out of the exception? Alternatively, you might mess around with `RestClient` to get it to do something more convenient.